tests/cpp_unit_tests/test_three_phase_tensor.cpp

```diff
@@ -85,7 +85,7 @@ TEST_CASE("Three phase tensor") {
         CHECK((mat4 == mat).all());
         CHECK(diag_mult(1.0, 2.0, 3.0) == 6.0);
         // test layout
-        double* arr = (double*)&mat1;
+        auto* arr = reinterpret_cast<double*>(&mat1);
```

Comment:
so `static_cast` wasn't working here?

Reply:
It was giving "invalid type conversion" error

Reply:
please have a look at https://stackoverflow.com/questions/14603546/invalid-type-conversion-using-static-cast-what-proper-casting-should-i-use .

you can try and see if you can write it like

```cpp
        // double* arr = (double*)&mat1;  // remove this line
        CHECK(static_cast<double>(mat[0]) == 1.0);
```

if it doesn't work, please revert this change and we'll need to revisit this later because I honestly don't know why we're doing it this way.

in particular, this is exactly the reason why we should use `static_cast` where possible and `reinterpret_cast` only where absolutely necessary (here, it is not absolutely necessary but it hides the fact that we do this because of the C-style cast).